### ISSUE TITLE
关于Lsk论文中Fair1mV1.0结果复现的一些问题（Some questions about the reproduction of Fair1mV1.0 results in the Lsk paper）

### ISSUE THREAD

**Issue by jsxyhelu** [zcablii/LSKNet#22]
### What's the feature?

作者你好，我一直在关注ISPRS（Fair1m）的比赛，在你的论文中记录LSK可以在Fair1mV1.0上达到 map 47.87，我在ISPRS的网站上也查到yuxuanli(推测就是你的账号）的得分为47.3128，这对于单模型来说应该是非常好的成绩。
之前我一直在做Fair1mV2.0的训练，也尝试过LSK，可能是因为参数选择的原因，对于Fair1mV2.0来说（数据集的处理方式是全部转换为jpg，而后采用1024 200进行切割；训练的时候是单卡A40，把synbn改为bn，lr用的还是0.001)在12epoch下最好的结果是map:0.36,在ISPRS上评分只有14.9618。
我认为这个结果没有显示出lsk的实际能力，现在准备开始在Fair1mV1.0下复现你的论文结果，包括竞赛结果。因为这里可能涉及到比较多的问题，所以想先了解一下：
1、数据集方面：应该也是使用1024 200的切割吧？有无进行格式转化（对于Fair1m数据集中提出的tif文件带地理空间信息我也研究过，似乎对于结果没有提高）。特别是Fair1mV1.0它没有区分训练集和验证集，那么你是如何划分的？
2、模型训练方面：是否就是使用lsk_s_backbone-e9d2e551.pth作为预训练模型？还是使用其它模型?达到 map 47.87需要训练多少epoch?
3、结果提交方面：我发现最后合理使用NMS能够对于ISPRS上分数提升有所帮助，请问是否有相关考虑？
最后，我发现lsk_s_backbone-e9d2e551.pth是放在mmlab的网站上面的，但是在mmrotate v1.0 中没有找到lsk的config，想了解一下相关情况。
非常感谢，主要还是想复现，回头如果可以我把这里具体的过程总结出来，并且继续在Fair1mV2.0上进行尝试。

=====Since it's on GitHub, write another English version==============================
Hello author, I have been following the ISPRS (Fair1m) competition, in your paper it is recorded that LSK can reach map 47.87 on Fair1mV1.0, and I also found on the ISPRS website that yuxuanli (presumably your account) has a score of 47.3128。it should be a very good score for a single model.
I have been doing Fair1mV2.0 training before, and I have also tried LSK, probably because of parameter selection, for Fair1mV2.0 (the processing method of the dataset is all converted to jpg, and then 1024 200 is used for cutting; When training is a single card A40, change synbn to bn, lr is still 0.001) The best result at 12epoch is map:0.36, and the rating on ISPRS is only 14.9618.
I  think this result HASNOT shows lsk's actual ability.Now I am ready to start reproducing your paper results, including competition results, under Fair1mV1.0. Because there may be more problems involved here, I want to understand it first:

1.  data set: it should also use 1024 200 cutting, right? There is no format conversion (I have also studied the geospatial information of the tif file proposed in the Fair1m dataset, and it seems that the results have not improved). 
In particular, Fair1mV1.0 does not distinguish between training set and validation set, so how do you divide it?

2. Model training: is it to use lsk_s_backbone-e9d2e551.pth as a pre-training model? Or use a different model? How much epoch do I need to train to reach MAP 47.87?

3.result submission: I find that the reasonable use of NMS can help improve the score on ISPRS, are there any relevant considerations?

Finally, I found that lsk_s_backbone-e9d2e551.pth is placed on the MMLAB website, but I can't find lsk's config in mmrotate v1.0.

Thank you very much, mainly I want to reproduce, go back if I can summarize the specific process here, and continue to try on Fair1mV2.0.

**Comment by zcablii** [comment]
Submitting results on ISPRS overwrites previous results, so the previous 47.87 results were overwritten by other test results submitted later at the time. But with the ckpts I provided, it should be easy to reproduce the 47.87 results.
My answer to your question is as follows:
1. The data processing for fair1mv1, as explained in the supplementary materials to the paper, is consistent with the default processing for dota. The method used is 1024 500 (overlap). I trained the model by converting the fair1m dataset into the form of dota dataset, then trained the model in the same way as the dota dataset, and finally converted the detection results into the format supported by the fair1m test server for submission. By training 12ep on the training set and then testing (no val set, consistent with dota).
2. Loading only the pre-trained backbone sk_s_backbone-e9d2e551.pth, then finetune the whole detection framework for 12 epochs.(consistent with dota experiments)
3. I haven't tried this part of the parameter for experimentation, I used the default NMS.
Finally, the code for mmrotate is placed at https://github.com/open-mmlab/mmrotate/tree/dev-1.x/projects/LSKNet

**Comment by zcablii** [comment]
Moreover, [https://github.com/zcablii/LSKNet/tree/main/tools/data/fair](url) here provides the tools for fair-dota conversion and data splitting.

**Comment by jsxyhelu** [comment]
Understand, I will start with 1024 500 (overlap) first！Thanks~

**Comment by zcablii** [comment]
One more thing just reckons me. When using a single GPU with bs=2, lr should make an adaptation accordingly. You could try `optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)` or 
`optimizer = dict(_delete_=True, type='AdamW', lr=4e-5, betas=(0.9, 0.999), weight_decay=0.05)`. lr = 0.001 is definitely too large for you setting.

**Comment by jsxyhelu** [comment]
明白，我正在做基于单个GPU的复现，给的信息非常有用的，感谢！
=============================================
Understand, I'm doing a single GPU-based reproduction, the information given is very useful, thanks!

**Comment by Balabala-Hong** [comment]
> 明白，我正在做基于单个GPU的复现，给的信息非常有用的，感谢！ Understand, I'm doing a single GPU-based reproduction, the information given is very useful, thanks!

Hello,  did you find suitable hyper-parameters in single or two machine? I'm facing the same problem in reproduceing the experimental results.
